openforcefield/tests/test_molecule.py

```diff
@@ -372,6 +372,59 @@ def test_to_networkx(self, molecule):
         """Test conversion to NetworkX graph."""
         graph = molecule.to_networkx()
 
+    @pytest.mark.parametrize('molecule', mini_drug_bank())
+    def test_to_inchi(self, molecule):
+        """Test conversion to standard and non-standard InChI"""
+        inchi = molecule.to_inchi()
+        non_standard = molecule.to_inchi(True)
+
+    @pytest.mark.parametrize('molecule', mini_drug_bank())
+    def test_to_inchikey(self, molecule):
+        """Test the conversion to standard and non-standard InChIKey"""
+        inchikey = molecule.to_inchikey()
+        non_standard_key = molecule.to_inchikey(True)
+
```

Comment:
(Blocking) Would we expect a OFFMol to survive an InChI roundtrip? The interaction of our chosen aromaticity model/kekulization scheme and InChI's information content is complicated, and we'd need something like a mathematical proof to know whether were safe here. But, I'm not that smart, so if we DO expect an OFFMol to survive an InChI roundtip, **we should make roundtrip tests using MiniDrugBank for both toolkits here**. Our standard for "survive a roundtrip" could just be `is_isomorphic_with`.

Reply:
Good point I think from reading the spec of standard InChI we would expect some information to be lost. This has been extensively tested by Openeye [here](https://docs.eyesopen.com/toolkits/python/oechemtk/OEChemFunctions/OEMolToInChI.html#OEChem::OEMolToInChI) and in the round trip testing, I have used non-standard InChi with a fixed hydrogen layer which seems to give the best results. We probably want to keep an eye on the test as some current issues could be fixed in future if requirements around nitrogen stereochemistry were changed.